spec/lib/stellar/transaction_builder_spec.rb

```diff
@@ -0,0 +1,282 @@
+        )
+      }.to raise_error(
+        ArgumentError, "bad :time_bounds"
+      )
+    end
+    it "bad base_fee" do
+      expect {
+        Stellar::TransactionBuilder.new(
+          source_account: key_pair,
+          sequence_number: 1,
+          base_fee: 0
+        )
+      }.to raise_error(
+        ArgumentError, "bad :base_fee"
+      )
+    end
+    it "bad memo" do
+      expect {
+        Stellar::TransactionBuilder.new(
+          source_account: key_pair,
+          sequence_number: 1,
+          memo: {"data" => "Testing bad memo"}
+        )
+      }.to raise_error(
+        ArgumentError, "bad :memo"
+      )
+    end
+    it "success" do
+      builder = Stellar::TransactionBuilder.new(
+        source_account: key_pair,
+        sequence_number: 1,
+        time_bounds: Stellar::TimeBounds.new(min_time: 0, max_time: 600),
+        base_fee: 200,
+        memo: "My test memo"
+      )
+      expect(builder.memo).to eql(Stellar::Memo.new(:memo_text, "My test memo"))
+    end
+  end
+
+  describe ".add_operation" do
+    it "bad operation" do
+      expect { 
+        builder.add_operation(
+          [:bump_sequence, 1]
+        )
+      }.to raise_error(
+        ArgumentError, "bad operation"
+      )
+    end
+
+    it "returns self" do
+      builder.add_operation(
+          Stellar::Operation.bump_sequence({"bump_to": 1})
+      ).add_operation(
+          Stellar::Operation.bump_sequence({"bump_to": 1})
+      )
+    end
+  end
+
+  describe ".clear_operations" do
+    it "can clear operations" do
+      builder = builder.add_operation(
+          Stellar::Operation.bump_sequence({"bump_to": 1})
+      ).clear_operations
+      expect(builder.operations).to eql([])
+    end
+
+    it "returns self" do
+      builder.add_operation(
+          Stellar::Operation.bump_sequence({"bump_to": 1})
+      ).clear_operations.add_operation(
+          Stellar::Operation.bump_sequence({"bump_to": 1})
+      )
+    end
+  end
+
+  describe ".set_sequence_number" do
+    it "allows sequence number to be updated" do
+      builder.set_sequence_number(5)
+      expect(builder.sequence_number).to eql(5)
+    end
+
+    it "returns self" do
+      builder.set_sequence_number(3).sequence_number
+    end
+
+    it "raises an error for bad sequence number" do
+      expect {
+        builder.set_sequence_number(nil) 
+      }.to raise_error(
+        ArgumentError, "bad sequence number"
+      )
+    end
+  end
+
+  describe ".set_timeout" do
+    it "raises an error for non-integers" do
+      expect {
+        builder.set_timeout(nil)
+      }.to raise_error(
+        ArgumentError, "timeout must be a non-negative integer"
+      )
+    end
+
+    it "raises an error for negative timeouts" do
+      expect {
+        builder.set_timeout(-1)
+      }.to raise_error(
+        ArgumentError, "timeout must be a non-negative integer"
+      )
+    end
+
+    it "returns self" do
+      builder.set_timeout(10).time_bounds
```

Comment:
This test says it returns self, but the test doesn't check what is returned. Is the test incomplete?

Reply:
Its not explicitly testing the return type but its testing that it has a `time_bounds` attribute, which I guess `Transaction` also has. I'll make it more explicit.